### PR TITLE
[Bugfix] Fix the load of loss function in Automm

### PR DESCRIPTION
**Pull request by Linuxdex**
*Issue #, if available:*
The loss function cannot be loaded properly in automm. It will affect the label transform in process_dataframe.py.
*Description of changes:*
The loss function will be used in the transform prediction. It couldn't be called after loading the model.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1819-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1819/1/index.html

**Comment by szha** [comment]
Job PR-1819-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1819/3/index.html

File: multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py [hunk 1]
```diff
@@ -441,7 +441,7 @@ def transform_prediction(
         Predicted labels ready to compute metric scores.
         """
         if loss_func is not None and isinstance(loss_func, torch.nn.BCEWithLogitsLoss):
-            y_pred = torch.sigmoid(y_pred)
+            y_pred = torch.sigmoid(y_pred.float())
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py:444]: We should revise the logic and add sigmoid in https://github.com/awslabs/autogluon/blob/bf995273bd16ed663b75e583feaf64f80b27e03c/multimodal/src/autogluon/multimodal/optimization/lit_module.py#L247

File: multimodal/src/autogluon/multimodal/optimization/lit_module.py [hunk 2]
```diff
@@ -148,7 +148,7 @@ def _compute_metric_score(
         label: torch.Tensor,
     ):
         if isinstance(self.loss_func, nn.BCEWithLogitsLoss):
-            logits = torch.sigmoid(logits)
+            logits = torch.sigmoid(logits.float())
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/optimization/lit_module.py:151]: Same here
